### ISSUE TITLE
MacOS Build Instructions are incomplete

### ISSUE THREAD

**Issue by notgif** [yvt/openspades#1012]
The current build instructions are (roughly):

1. get xcode tools
2. bootstrap vcpkg
3. vcpkg install
4. create build dir & build

### VCPKG
I needed to initialise the vcpkg submodule explicitly (with) ``` git submodule update --init --recursive vcpkg ``` and then checkout the submodule to the vcpkg master branch—the one pointed to in this repo couldn't build a certain package, or detect my gcc version (gcc-11).

### Compiler
Also, on MacOS versions earlier than (<) 10.15 vcpkg complains about the xcode-supplied clang compiler, and you need to install gcc itself.

### Xcode
I was able to build successfully with only the xcode command line tools, so ```sudo xcodebuild -license``` doesn't necessarily need to be run. It requires a full xcode install, which is a fairly steep requirement ( a full xcode install is uncomfortable large ).

I'll submit a pull request with an updated process, but I'm not actually sure I've found the optimal solution (especially with the git commands). I also haven't used Nix, so I can't verify if/what commands are necessary for using that for packages.

**Comment by yvt** [comment]
> I needed to initialise the vcpkg submodule explicitly (with) `git submodule update --init --recursive vcpkg`

The submodule can be initialized at clone time by `git clone ... --recurse-submodules` (as noted in the Windows build instructions). Or did this not work?

> checkout the submodule to the vcpkg master branch—the one pointed to in this repo couldn't build a certain package, or detect my gcc version (gcc-11).

Can you send a separate PR to update the vcpkg submodule? This submodule is intended to serve a role similar to `package-lock.json` and `Cargo.lock` and not to be updated arbitrarily in a local (`git clone`-ed) working tree.

> I also haven't used Nix, so I can't verify if/what commands are necessary for using that for packages.

I can check that part.

**Comment by notgif** [comment]
> The submodule can be initialized at clone time...

`git clone .. --recurse-submodules` should work (I'll check bc _ya never know_) — I just only looked through the Linux and Mac sections and didn't think to look through the Windows bit.

> Can you send a separate PR to update the vcpkg submodule? ...

Yeah I can do that, should I create a separate issue for it as well?
I think it was microsoft/vcpkg#24172 that I came across after I tried manually patching `vcpkg` to find my `gcc-11`. And the solutions in that issue were just 'update'.

> Nix

Easy, I'll put a draft part in for that
